docs/tutorials/docker.md

```diff
@@ -0,0 +1,81 @@
+
+Before beginning the quickstart, it is helpful to read the [general Druid overview](../design/index.md) and the
+[ingestion overview](../ingestion/index.md), as the tutorials will refer to concepts discussed on those pages. Additionally, familiarity with Docker is recommended.
+
+## Prerequisites
+
+* Docker
+
+## Getting started
+
+The Druid source code contains [an example `docker-compose.yml`](https://github.com/apache/druid/blob/master/distribution/docker/docker-compose.yml) which can pull an image from Docker Hub and is suited to be used as an example environment and to experiment with Docker based Druid configuration and deployments.
+
+### Compose file
+
+The example `docker-compose.yml` will create a container for each Druid service, as well as Zookeeper and a PostgreSQL container as the metadata store. Deep storage will be a local directory, by default configured as `./storage` relative to your `docker-compose.yml` file, and will be mounted as `/opt/data` and shared between Druid containers which require access to deep storage. The Druid containers are configured via an [environment file](https://github.com/apache/druid/blob/master/distribution/docker/environment).
+
+### Configuration
+
+Configuration of the Druid Docker container is done via environment variables, which may additionally specify paths to [the standard Druid configuration files](../configuration/index.md)
+
+Special environment variables:
+
+* `JAVA_OPTS` -- set java options
+* `DRUID_LOG4J` -- set the entire `log4j.xml` verbatim
+* `DRUID_LOG_LEVEL` -- override the default log level in default log4j
+* `DRUID_XMX` -- set Java `Xmx`
+* `DRUID_XMS` -- set Java `Xms`
+* `DRUID_MAXNEWSIZE` -- set Java max new size
+* `DRUID_NEWSIZE` -- set Java new size
+* `DRUID_MAXDIRECTMEMORYSIZE` -- set Java max direct memory size
+* `DRUID_CONFIG_COMMON` -- full path to a file for druid 'common' properties
+* `DRUID_CONFIG_${service}` -- full path to a file for druid 'service' properties
+
+In addition to the special environment variables, the script which launches Druid in the container will also attempt to use any environment variable starting with the `druid_` prefix as a command-line configuration. For example, an environment variable
+ 
+```druid_metadata_storage_type=postgresql```
+ 
+would be translated into 
+
+```-Ddruid.metadata.storage.type=postgresql```
+ 
+for the Druid process in the container.
+
+The Druid `docker-compose.yml` example utilizes a single environment file to specify the complete Druid configuration, however in production use cases we suggest using either `DRUID_COMMON_CONFIG` and `DRUID_CONFIG_${service}` or specially tailored service specific environment files.
+
+## Launching the cluster
+
+Run `docker-compose up` to launch the cluster will a shell attached, or `docker-compose up -d` to run the cluster in the background. Once the cluster has started, you can navigate to [http://localhost:8888](http://localhost:8888).
```

Comment:
"cluster will" -> " cluster with"

Reply:
Also, maybe others would know, because the path is in a section above, but it wasn't obvious to me to run the command from druid_repo_home/distribution/docker. So optionally, could mention, i.e., "Run `docker-compose up` from the distribution/docker directory to launch...."

One more thing, do they need to do the docker commands that are in the druid_repo_home/distribution/docker/README.md ? I did, but wasn't sure I had to.

Reply:
Fixed typo.

The steps in the readme are only necessary if building druid image from source, which I'm not sure needs to be mentioned here.